### ISSUE TITLE
I can no longer access terminal accessibility buffer with orca

### ISSUE THREAD

**Issue by jvesouza** [microsoft/vscode#173532]
Type: <b>Bug</b>

CC: @meganrogge 
CC: @isidorn 
Today I updated the VScode insiders version and noticed that it is no longer possible to navigate with arrows in the terminal accessibility buffer.

I create a new terminal, type any command, press shift+tab and try to use the arrow keys to read the result of the command. Nothing is read by orca, no matter what keys I use.
I also tested disabling all extensions and the result was the same.

VS Code version: Code - Insiders 1.76.0-insider (c9e3ef865539061b5602c6b9ac6030332aa1ce70, 2023-02-06T05:24:21.999Z)
OS version: Linux x64 6.1.9-200.fc37.x86_64
Modes:
Sandboxed: Yes

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-8250U CPU @ 1.60GHz (8 x 3000)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|1, 1, 1|
|Memory (System)|15.27GB (10.91GB free)|
|Process Argv|-n --crash-reporter-id bc02adbd-6c3a-44b8-af92-55103a608af7|
|Screen Reader|yes|
|VM|0%|
|DESKTOP_SESSION|gnome-xorg|
|XDG_CURRENT_DESKTOP|GNOME|
|XDG_SESSION_DESKTOP|gnome-xorg|
|XDG_SESSION_TYPE|x11|
</details><details><summary>Extensions (37)</summary>

Extension|Author (truncated)|Version
---|---|---
doxdocgen|csc|1.4.0
vscode-markdownlint|Dav|0.49.0
vscode-chatgpt|gen|3.5.5
rest-client|hum|0.25.1
better-cpp-syntax|jef|1.17.2
azure-pipelines|ms-|1.208.0
vscode-docker|ms-|1.23.3
csharp|ms-|1.25.4
vscode-kubernetes-tools|ms-|1.3.11
remote-containers|ms-|0.276.0
remote-ssh|ms-|0.97.2023020215
remote-ssh-edit|ms-|0.84.0
azure-account|ms-|0.11.3
cmake-tools|ms-|1.13.45
cpptools|ms-|1.14.1
cpptools-extension-pack|ms-|1.3.0
makefile-tools|ms-|0.6.0
powershell-preview|ms-|2023.2.0
remote-explorer|ms-|0.2.0
vsliveshare|ms-|1.0.5828
fabric8-analytics|red|0.3.6
java|red|1.15.2023020403
vscode-xml|red|0.25.2023020103
vscode-yaml|red|1.11.0
sonarlint-vscode|Son|3.14.0
cmake|twx|0.0.17
intellicode-api-usage-examples|Vis|0.2.7
vscodeintellicode|Vis|1.2.30
vscode-boot-dev-pack|vmw|0.2.1
vscode-spring-boot|vmw|1.44.0
vscode-java-debug|vsc|0.48.0
vscode-java-dependency|vsc|0.21.2023013100
vscode-java-pack|vsc|0.25.2022112900
vscode-java-test|vsc|0.37.2023013002
vscode-maven|vsc|0.40.2023020103
vscode-spring-boot-dashboard|vsc|0.10.1
vscode-spring-initializr|vsc|0.11.2

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492cf:30256198
vstes627cf:30244335
vslsvsres303:30308271
pythonvspyl392:30422396
pythontb:30258533
pythonptprofiler:30281269
vshan820:30294714
pythondataviewer:30285072
vscod805cf:30301675
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593cf:30376535
pythonvs932:30404738
cppdebug:30492333
vsclangdf:30492506
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30618038
pylantcb52:30590116
pyindex848:30611229
nodejswelcome1:30587009
pyind779:30611226
pythonsymbol12:30651887
hidelangstatbar:30657380

```

</details>

**Comment by isidorn** [comment]
@meganrogge assigning to you first - can you investigate?
Did something change on the Terminal side with latest Insiders?

fyi @Tyriar

**Comment by meganrogge** [comment]
Related #173452

**Comment by meganrogge** [comment]
I can reproduce. the terminal underneath is getting the focus / mouse /keyboard events

**Comment by meganrogge** [comment]
I cannot reproduce the issue in NVDA with the steps here. I haven't tried with Orca - can you see if it works for you with NVDA?

https://github.com/microsoft/vscode/issues/173452#issuecomment-1419093065

**Comment by jvesouza** [comment]
@meganrogge Thanks for the reply.
Something must have changed since Thursday because until then I was able to use the arrows in the terminal's accessibility buffer.
Would @joanmarie be able to help?

**Comment by meganrogge** [comment]
@jvesouza what changed is we no longer use a content editable text area element - due to the issues reported with that

**Comment by meganrogge** [comment]
You should be able to navigate it just like any other web document using the keys you'd typically use on a web page with orca

**Comment by Tyriar** [comment]
@meganrogge we should get an orca setup to test this out. It would be unfortunate, but maybe `'auto'` in the content editable setting needs to also take `isLinux` into account?

**Comment by jooyoungseo** [comment]
Let me chime in. @meganrogge and @Tyriar -- Currently, "document" role does not force some screen readers' (e.g., JAWS and ORCA) virtual cursor automatically. NVDA and VoiceOver work properly though. Interestingly, JAWS and ORCA work when tabindex="0" is assigned to the div where "document" role resides. I can take a deep look when I have more time later.

**Comment by jooyoungseo** [comment]
@meganrogge Under `.xterm-accessibility-tree`, each listitem is assigned tabindex "-1". My hunch is that this interrupts terminal buffer document role tabindex. Why don't we try removing the listitem's tabindex attribute?

**Comment by meganrogge** [comment]
@jooyoungseo thank you so much for that helpful info. 

the `.xterm-accessibility-tree` was removed with https://github.com/microsoft/vscode/pull/173322 . @jvesouza could you pls test in OSS off of main with Orca to see if it's fixed?

**Comment by jooyoungseo** [comment]
Alright. Here is my investigation results. JAWS, Narrator, ORCA users need to set `{"terminal.integrated.accessibleBufferContentEditable": "on"}` whereas NVDA and VoiceOver users can just keep the "auto" or "off" contenteditable settings. I am not quite sure why JAWS, Narrator, and ORCA do not switch on their virtual cursor against document role while NVDA and VoiceOver do their jobs. @meganrogge -- Do you have any connection with Narrator team to further investigate this issue?

**Comment by meganrogge** [comment]
Thank you for investigating and confirming that. I think @isidorn would be the best one to answer that question.

>Do you have any connection with Narrator team to further investigate this issue?

I wonder if we have a way to detect which screen reader is used. For now, as @Tyriar suggests, we should set `{"terminal.integrated.accessibleBufferContentEditable": "on"}` for linux users

**Comment by jvesouza** [comment]
Confirmed, with the {"terminal.integrated.accessible Buffer ContentEditable": "on"}, I can navigate with the arrows in the terminal accessibility buffer using orca.

I also tested using nvda and with nvda everything worked very well, without me having to make any changes.

One difference I noticed between orca and nvda is that using nvda the position in the buffer is always preserved when I switch between the terminal and the accessibility buffer. Now with orca, whenever I press shift+tab, the focus always seems to be moved to the beginning of the accessibility buffer.
Because of this, we will have to use the arrows to find the desired content.

**Comment by lramos15** [comment]
@meganrogge How can I verify this do I need to have an orca setup?

@jvesouza Would you mind testing this in the latest insiders to see if it works as expected?

**Comment by VSCodeTriageBot** [comment]
This bug has been fixed in the latest release of [VS Code Insiders](https://code.visualstudio.com/insiders/)!

@jvesouza, you can help us out by commenting `/verified` if things are now working as expected.

If things still don't seem right, please ensure you're on version d0acdbff1f30eac7cc68930013485276d1bae0ec of Insiders (today's or later - you can use `Help: About` in the command palette to check), and leave a comment letting us know what isn't working as expected.

Happy Coding!

**Comment by jvesouza** [comment]
> This bug has been fixed in the latest release of [VS Code Insiders](https://code.visualstudio.com/insiders/)!
> 
> @jvesouza, you can help us out by commenting `/verified` if things are now working as expected.
> 
> If things still don't seem right, please ensure you're on version [d0acdbf](https://github.com/microsoft/vscode/commit/d0acdbff1f30eac7cc68930013485276d1bae0ec) of Insiders (today's or later - you can use `Help: About` in the command palette to check), and leave a comment letting us know what isn't working as expected.
> 
> Happy Coding!



> @meganrogge How can I verify this do I need to have an orca setup?
> 
> @jvesouza Would you mind testing this in the latest insiders to see if it works as expected?

Yes, it is working perfectly.
